src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts

```diff
@@ -263,11 +263,15 @@ export const retentionAnalysisVisualColumns: InputColumn[] = [
 
 export const createDataSet = async (quickSight: QuickSight, awsAccountId: string, principalArn: string,
   dataSourceArn: string,
-  props: DataSetProps)
+  props: DataSetProps, requestAction: ExploreRequestAction)
 : Promise<CreateDataSetCommandOutput|undefined> => {
 
   try {
-    const datasetId = `${QUICKSIGHT_TEMP_RESOURCE_NAME_PREFIX}${uuidv4()}`;
+    let datasetId = uuidv4();
```

Comment:
If requestAction is `Publish`, please use `${QUICKSIGHT_RESOURCE_NAME_PREFIX}${QUICKSIGHT_DATASET_INFIX}${uuidv4().replace(/-/g, '')}`